src/libcore/cell.rs

```diff
@@ -187,6 +187,29 @@ use ops::{Deref, DerefMut, CoerceUnsized};
 use ptr;
 
 /// A mutable memory location.
+/// 
+/// ```
+/// use std::cell::Cell;
+///
+/// struct SomeStruct {
+///     regular_field: u8,
+///     special_field: Cell<u8>,
+/// }
+///
+/// let my_struct = SomeStruct {
+///     regular_field: 0,
+///     special_field: Cell::new(1),
+/// };
+///
+/// let new_value = 100;
+/// 
+/// // ERROR, because my_struct is immutable
+/// // immutable.regular_field = new_value;
+/// 
+/// // WORKS, special_field is mutable because it is Cell
+/// immutable.special_field.set(new_value);
+/// assert_eq!(immutable.special_field.get(), new_value);
+/// ```
```

Comment:
I've added some prose and example section.
BTW: I'd see reference to this blog post:
https://ricardomartins.cc/2016/06/08/interior-mutability
in the docs. but perhaps adding links to blog post is not the way of doing docs. :)

Maybe it's just me but I was reading this https://doc.rust-lang.org/std/cell/ a few times and still didn't know what `Cell` is all about then read the blog post one to get the concept easily.. .